lib/pos_blockchain.go

```diff
@@ -507,28 +587,43 @@ func (bc *Blockchain) upsertBlockAndBlockNodeToDB(desoBlock *MsgDeSoBlock, newBl
 	return nil
 }
 
+func (bc *Blockchain) putBestHash(hash *BlockHash) error {
```

Comment:
Nit: rename this for clarity. I realize that the existing naming uses the "put best hash" term, but this seems like a decent place to break that so the call sites for this function are clearer
```suggestion
func (bc *Blockchain) storeTipHashInDB(hash *BlockHash) error {
```